.github/CONTRIBUTING.md

```diff
@@ -1,8 +1,66 @@
+
+The plugins directory has one folder per plugin. To create the correct directory structure for your plugin, use [@blockly/create-package](https://www.npmjs.com/package/@blockly/create-package).
+
+## How can I contribute?
+
+- [Suggest](https://github.com/google/blockly-samples/issues/new?assignees=&labels=type%3A+feature+request%2C+triage&template=feature_request.md) plugins, examples, or codelabs
+- Implement plugins, examples, or codelabs
+- [Report bugs](https://github.com/google/blockly-samples/issues/new?assignees=&labels=type%3A+bug%2C+triage&template=bug_report.md)
+- Fix bugs
+- Reproduce bugs
+
+## How do I add a plugin?
+
+Plugins go through four stages: suggestion, discussion, implementation, and publishing:
+
+A plugin starts as a **suggestion**. You can suggest a plugin by creating a new issue with the [Feature Request](https://github.com/google/blockly-samples/issues/new?assignees=&labels=type%3A+feature+request%2C+triage&template=feature_request.md) template. To write a good suggestion:
+- Describe the desired functionality.
+- Describe the API the plugin would expose.
+- List alternatives you've considered.
+- List any APIs that need to be added or changed in core Blockly to support the plugin.
+- Include screenshots, GIFs, or mock-ups if the plugin includes UI features.
+- Explain why it should be a first-party plugin rather than a third-party plugin.
+
+Before making a suggestion, [search](https://github.com/google/blockly-samples/issues?q=is%3Aopen+is%3Aissue+label%3A%22category%3A+plugin%22+label%3A%22type%3A+feature+request%22) for a related issue. If you find one, add a comment to the existing issue instead of opening a new one.
+
+Next a plugin will go into the **discussion** phase. This phase includes:
+- Clarification of the desired functionality.
+- Clarification of the plugin's API.
+- Planning for implementation.
+- Planning for tests.
+- Discussion of API changes in core Blockly.
+- Breaking large plugins into implementation steps.
+- Plugin naming, based on naming conventions that we will publish Real Soon Now.
+
+The goal is that at the end of the discussion phase all major design decisions have been made and there is a clear list of implementation steps. During discussion we may decide that a plugin should be a third-party plugin, and not be published under the `@blockly` scope. In that case we will close the issue.
+
+Next is the **implementation** phase, which includes
+- Creating the plugin's folder.
+- Implementing core logic for the plugin.
+- Implementing a UI, if needed.
+- Adding tests.
+- Writing the plugin's `README.md`.
+
+Implentation may be done by multiple contributors in parallel.
+
+Finally, **publishing**. This is done by Blockly team members when implementation is complete.
```

Comment:
Not for this PR, but we'll want to expand this section on publishing cadence that developers can expect.